### ISSUE TITLE
Update Google Material icons

### ISSUE THREAD

**Issue by Nek-12** [mikepenz/Android-Iconics#619]
## About this issue

I'm using Google Rounded material and I'm frequently encountering missing icons compared to [here](https://fonts.google.com/icons?selected=Material+Icons:mic&icon.query=interests&icon.style=Rounded) (the link is an example of a missing icon)

## Details
- [x] Used library version Google rounded material 4.0.0.1-kotlin@aar

## Checklist

- [x] Searched for [similar issues](https://github.com/mikepenz/Android-Iconics/issues)
- [x] Checked out the [sample application](https://github.com/mikepenz/Android-Iconics/tree/develop/app)
- [x] Read the [README](https://github.com/mikepenz/Android-Iconics/blob/develop/README.md)
- [x] Checked out the [CHANGELOG](https://github.com/mikepenz/Android-Iconics/releases)
- [x] Read the [FAQ](https://github.com/mikepenz/Android-Iconics/blob/develop/FAQ.md)
- [x] Checked out the [MIGRATION GUIDE](https://github.com/mikepenz/Android-Iconics/blob/develop/MIGRATION.md)

**Comment by mikepenz** [comment]
The latest version here is still v4.0.0: https://github.com/google/material-design-icons/releases

**Comment by Nek-12** [comment]
I believe it's just the "scrapper" version. The actual icons are reflected on the API. 
Otherwise I don't really know how to explain the discrepancy in icon availability

**Comment by dummyco** [comment]
Yeah, it seems like Google is migrating towards "Material Symbols" with a lot of new and changed icons and new styling principles (Fill/Weight/Grade/Optical size).
For now, I've created a custom Iconics font/mapping for filled and outlined variants from the library contents (via parsing names and building font), but, probably, that would be super cool to provide one from Iconics.

refs:

https://fonts.google.com/icons
https://github.com/google/material-design-icons

**Comment by mikepenz** [comment]
Thank you @dummyco for chiming in on it. 

There doesn't seem to be a provided packaged ttf anymore with the css mapping, unless I am overlooking this one?

**Comment by dummyco** [comment]
@mikepenz yep, that was a headache 😄 
I've ended up cloning the material icons repo, writing the script to walk through `material-design-icons/symbols/web/` directory, filtering `materialsymbolsoutlined` + `_wght500grad200_48px`, `_wght500grad200fill1_48px` variants and feeding them into this [fantasticon](https://github.com/tancredi/fantasticon) tool (to generate ttf+css mapping) and in your online project generation tool next.

One more thing, it seems like there are too many icons to contain them in a single set of fields (enum), so there was also a need to break for filled and non-filled enums.

**Comment by mikepenz** [comment]
@dummyco yeah that's the most annoying piece for those super large icon sets. Having the same issue for `CommunityMaterial`: https://github.com/mikepenz/Android-Iconics/blob/develop/community-material-typeface-library/src/main/java/com/mikepenz/iconics/typeface/library/community/material/CommunityMaterial.kt#L81
splitting it into 3 enums. 


I was investigating at some point to use other constructs but they were even worse in size and classes. 
E.g. sealed class, ... 


The only "solution" would be to loose type safety and drop the enum, and only have a list of items, and you'd need to use a string to reference icons (which in my opinion would probably be worse) 

Also with the different variants having to maintain: https://github.com/mikepenz/Android-Iconics/tree/develop/google-material-typeface is quite some effort. 


On a relevant note. It is most likely better (given the material symbols) and the great vectorDrawable support these days (depending on your usecase) to get the specific items at the times they are needed. (and when not a huge amount of icons is needed)

**Comment by Nek-12** [comment]
> writing the script to walk through `material-design-icons/symbols/web/` directory, filtering `materialsymbolsoutlined` + `_wght500grad200_48px`, `_wght500grad200fill1_48px` variants and feeding them into this [fantasticon](https://github.com/tancredi/fantasticon) tool (to generate ttf+css mapping) and in your online project generation tool next.

Could you please share the script you used?

**Comment by dummyco** [comment]
Sure, I'll try to compose some user-friendly steps/snippet asap

**Comment by ViRGiL175** [comment]
Hi. I encountered the same problem. 

I'm not familiar with Compose technology, but there are convenient access to Material Design Icons via Compose classes: https://developer.android.com/reference/kotlin/androidx/compose/material/icons/Icons

It mentioned in official Google Fonts site. Looks like there are usual and extended icon sets, that already managed and updating by Google. 
<img width="1240" alt="image" src="https://user-images.githubusercontent.com/11541555/223474930-e1a1beb3-8817-4807-9036-99b08703e705.png">

Maybe there is any way to reuse this Compose icons in Android-Iconics library? It could be something like compatibility layer for Compose icons.

Or are there any plans to update `google-material-typeface` to latest icons set? 

We discovered your library recently and it looks very cozy. But due to lack of some icons it became question about using the library.

**Comment by Nek-12** [comment]
Those icons are basically shape/path classes with some fill. This library is about a different thing - using fonts as icon packs. 
If you want to use those icons, just use the relevant icon composable.
As for missing icons, a somewhat hacky solution is outlined above
